### DISCUSSION TITLE
Realtime performance impact

### DISCUSSION THREAD

**Discussion by guilhermeguidetti** [pocketbase/pocketbase#4257] [Q&A] [answered]
How many simultaneous connections can it handle for realtime updates before I might start seeing some poerformance issues? Just trying to figure out some scaling stuff for my app.

**Comment by ganigeorgiev** [comment, marked as answer]
This is difficult to answer as it will depend on a lot of factors - the server resources (more CPU will allow handling more realtime connections but it will also require more RAM because the client state remains in memory for the duration of each connection), the schema of the updated records (ex. how large is a single record), the amount of regular traffic and what your app actually does, how many of those realtime connections are idle, etc.

For write benchmarks you could check the tests from https://github.com/pocketbase/benchmarks (all are done on shared vCPU servers).

**Comment by dineshkp** [comment]
Hi @guilhermeguidetti,
I'm researching PocketBase for a real-time application and found your performance discussion helpful. Would you mind sharing:

1. **What scale are you running in production?** How many concurrent realtime connections do you typically handle?
2. **Server specs:** What hardware configuration (CPU/RAM) are you using and how does it perform under realtime load?
3. **Performance insights:** Any bottlenecks, optimizations, or unexpected behavior you've encountered with realtime connections?
4. **Resource planning:** How do you estimate server requirements for your connection count?

Since @ganigeorgiev mentioned performance depends on many factors, real-world data points from actual implementations would be incredibly valuable for capacity planning.
